packages/design/src/Popover/Transition.tsx

```diff
@@ -28,8 +34,6 @@ class Transition extends React.Component {
     delete childProps.onEntering;
 
     const child = React.Children.only(children);
-    return React.cloneElement(child, childProps);
+    return React.cloneElement(child as React.ReactElement<any>, childProps);
```

Comment:
Instead of casting this component should have children prop with type ReactElement
```tsx
interface TransitionProps {
  onEntering: (node: Element | Text | null) => void;
  children: React.ReactElement;
}

export class Transition extends React.Component<TransitionProps> {
  componentDidMount() {
...
```